src/test/java/teammates/test/cases/action/AdminActivityLogPageActionTest.java

```diff
@@ -1,27 +1,564 @@
 package teammates.test.cases.action;
 
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.TimeZone;
+
 import org.testng.annotations.Test;
 
+import teammates.common.util.ActivityLogEntry;
 import teammates.common.util.Const;
+import teammates.common.util.JsonUtils;
+import teammates.common.util.TimeHelper;
+import teammates.test.driver.FileHelper;
+import teammates.test.driver.TestProperties;
 import teammates.ui.controller.AdminActivityLogPageAction;
+import teammates.ui.controller.AjaxResult;
+import teammates.ui.controller.ShowPageResult;
+import teammates.ui.pagedata.AdminActivityLogPageData;
+import teammates.ui.pagedata.PageData;
+
+import com.google.gson.reflect.TypeToken;
 
 public class AdminActivityLogPageActionTest extends BaseActionTest {
 
+    private static final String TYPICAL_LOG_MESSAGE = "/typicalLogMessage.json";
+
+    private static final int LOG_MESSAGE_INDEX_TODAY = 0;
+    private static final int LOG_MESSAGE_INDEX_YESTERDAY = 1;
+    private static final int LOG_MESSAGE_INDEX_TWO_DAYS_AGO = 2;
+
+    private static final int LOG_MESSAGE_INDEX_MANY_LOGS = 3;
+
+    private static final int LOG_MESSAGE_INTERVAL_MANY_LOGS = 130;
```

Comment:
You may want to explain the magic numbers that may not be so obvious, e.g. this one (why 130?).